src/main/kotlin/com/cosmotech/connector/adt/impl/AzureDigitalTwinsConnector.kt

```diff
@@ -67,27 +67,30 @@ class AzureDigitalTwinsConnector : Connector<DigitalTwinsClient,List<CsvData>,Li
 
         modelInformationList = AzureDigitalTwinsUtil.retrievePropertiesFromBaseModels(modelInformationList)
 
-        LOGGER.info("Fetching Digital Twin Instances...")
+        LOGGER.info("Fetching Digital Twin Instances Information...")
         val fetchDTInstancesStart = System.nanoTime()
-        val digitalTwinInformation = constructDigitalTwinInstances(modelInformationList, client)
+
+        val digitalTwins = client.query("SELECT * FROM DIGITALTWINS", BasicDigitalTwin::class.java)
+            .sortedBy { it.metadata.modelId }
```

Comment:
I think the different calls to `sortedBy` could be expensive f we have a lot of twins. I suggest you remove all calls to `sortedBy`.
We have no constraint on the order of rows to generate in the CSV files. If we want to have some kind of determinism, it is still possible to sort the files in a separate CLI process.